src/sidebar/util/groups.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+/**
+ * Combine groups from multiple api calls together to form a unique list of groups.
+ * Add a listGroup property to each group identifying it as a "mygroup" or a "featured" group.
+ *
+ * @param {myGroups[]} groups the user is a member of
+ * @param {featuredGroups[]} groups scoped to the particular uri, authority, and user
```

Comment:
The syntax here is `@param {Type} name` for a required argument or `@param [Type] name` for an optional argument. So this would be `@param {Group[]} myGroups` for an array of groups called `myGroups`. We also follow the optional convention of adding a hyphen after the parameter name, before the description. See [JSDoc @param tag details](http://usejsdoc.org/tags-param.html).

The `Group` type isn't actually defined in our code currently, its just a name we've given to the structure of group objects returned from the API. We do the same with `Annotation`. If we did want to define it in our code somewhere that could be done with `@typedef`.